<?xml version="1.0" ?>
<robot name="complete">
	<link name="torso:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/torso_iam_fb80b350.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.026368 -0.075212 0.525308"/>
			<material name="torso:11_color">
				<color rgba="0.270588 0.270588 0.270588 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/torso_iam_fb80b350.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.026368 -0.075212 0.525308"/>
		</collision>
		<inertial>
			<mass value="0.607828"/>
			<inertia ixx="0.006338" ixy="0.000004" ixz="0.000248" iyy="0.005208" iyz="0.000001" izz="0.002162"/>
			<origin rpy="0 0 0" xyz="0.01632 -0.000121 0.649656"/>
		</inertial>
	</link>
	<link name="right_shoulder:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_shoulder_up_iam_34db21c8.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.062271 0.032843 -0.74045"/>
			<material name="right_shoulder:11_color">
				<color rgba="0.176471 0.176471 0.176471 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_shoulder_up_iam_34db21c8.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.062271 0.032843 -0.74045"/>
		</collision>
		<inertial>
			<mass value="0.084038"/>
			<inertia ixx="0.000026" ixy="-0.000001" ixz="-0.000002" iyy="0.000034" iyz="-0" izz="0.000033"/>
			<origin rpy="0 0 0" xyz="0.005637 0.000876 0.033153"/>
		</inertial>
	</link>
	<joint name="r_shoulder_z" type="revolute">
		<parent link="torso:11"/>
		<child link="right_shoulder:11"/>
		<origin rpy="0 0 -0.20444" xyz="0.026783 -0.049912 0.748809"/>
		<axis xyz="0 0 1"/>
		<limit effort="30" lower="-0.8" upper="0.8" velocity="1"/>
	</joint>
	<link name="right_collarbone:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_upper_arm_iam_e3e9c979.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.0285 -0.054827 -0.000102"/>
			<material name="right_collarbone:11_color">
				<color rgba="0.176471 0.176471 0.176471 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_upper_arm_iam_e3e9c979.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.0285 -0.054827 -0.000102"/>
		</collision>
		<inertial>
			<mass value="0.024811"/>
			<inertia ixx="0.000014" ixy="-0" ixz="0" iyy="0.000011" iyz="-0.000001" izz="0.000019"/>
			<origin rpy="0 0 0" xyz="-0 -0.041345 -0.002364"/>
		</inertial>
	</link>
	<joint name="r_shoulder_y" type="revolute">
		<parent link="right_shoulder:11"/>
		<child link="right_collarbone:11"/>
		<!-- <origin rpy="0 0 0.20444" xyz="0.026211 -0.018056 0.037"/> -->
		<!-- <origin rpy="0 0 0.20444" xyz="-0.025393 -0.020015 0.037"/> -->
		<origin rpy="0 0 0.20444" xyz="-0.021627 -0.02471212 0.037"/>
		<!-- left for comparison -->
		<!-- <origin rpy="0 -0.00445 -0.019538" xyz="0.022445 0.022567 0.037"/> -->
		<!-- <origin rpy="0 -0.00445 -0.019538" xyz="-0.021627 0.020528 0.037"/> -->
		<axis xyz="0 -1 0"/>
        <limit effort="30" lower="-3.142" upper="3.142" velocity="1"/>
	</joint>
	<link name="right_upper_arm:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_lower_arm1_iam_65a34910.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.044001 -0.021 -0.139458"/>
			<material name="right_upper_arm:11_color">
				<color rgba="0.176471 0.176471 0.176471 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_lower_arm1_iam_65a34910.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.044001 -0.021 -0.139458"/>
		</collision>
		<inertial>
			<mass value="0.189729"/>
			<inertia ixx="0.000851" ixy="0" ixz="0" iyy="0.000851" iyz="-0.000005" izz="0.000048"/>
			<origin rpy="0 0 0" xyz="-0.02306 -0.000321 -0.0866"/>
		</inertial>
	</link>
	<joint name="r_arm_x" type="revolute">
		<parent link="right_collarbone:11"/>
		<child link="right_upper_arm:11"/>
		<origin rpy="0 0 0" xyz="0.0245 -0.073664 -0.022853"/>
		<axis xyz="1 0 0"/>
		<limit effort="30" lower="-1.57" upper="0.314" velocity="1"/>
	</joint>
	<link name="right_lower_arm:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_wrist_iam_771ea33e.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.0168 -0.0018 -0.123873"/>
			<material name="right_lower_arm:11_color">
				<color rgba="0.176471 0.176471 0.176471 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_wrist_iam_771ea33e.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.0168 -0.0018 -0.123873"/>
		</collision>
		<inertial>
			<mass value="0.076793"/>
			<inertia ixx="0.000104" ixy="-0" ixz="-0" iyy="0.000095" iyz="-0.000002" izz="0.000021"/>
			<origin rpy="0 0 0" xyz="0 0.022485 -0.081055"/>
		</inertial>
	</link>
	<joint name="r_elbow_y" type="revolute">
		<parent link="right_upper_arm:11"/>
		<child link="right_lower_arm:11"/>
		<origin rpy="0 0 0" xyz="-0.023001 -0.024001 -0.1732"/>
		<axis xyz="0 1 0"/>
		<limit effort="30" lower="-1.745" upper="0.0" velocity="1"/>
	</joint>
	<link name="right_wrist:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_wrist_iam_81504746.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.000319 0.006617 -0.070181"/>
			<material name="right_wrist:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_wrist_iam_81504746.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.000319 0.006617 -0.070181"/>
		</collision>
		<inertial>
			<mass value="0.028397"/>
			<inertia ixx="0.000043" ixy="-0" ixz="0.000004" iyy="0.000058" iyz="-0.000007" izz="0.000029"/>
			<origin rpy="0 0 0" xyz="0.000031 -0.011056 -0.0675"/>
		</inertial>
	</link>
	<joint name="r_wrist_z" type="revolute">
		<parent link="right_lower_arm:11"/>
		<child link="right_wrist:11"/>
		<!-- <origin rpy="0 0 0.028946" xyz="0 0.0102 -0.12605"/> -->
		<origin rpy="0 0 0.028946" xyz="0 0.0102 -0.11105"/>
		<axis xyz="0 0 1"/>
		<limit effort="30" lower="-1.571" upper="1.571" velocity="1"/>
	</joint>
	<link name="right_palm:11">
		<visual>
			<geometry>
				<mesh filename="./meshes/right_palm_iam_583b22df.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.022933 0.001906 -0.017233"/>
			<material name="right_palm:11_color">
				<color rgba="0.74902 0.74902 0.74902 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<mesh filename="./meshes/right_palm_iam_583b22df.STL" scale="0.001 0.001 0.001"/>
			</geometry>
			<origin rpy="0 0 0" xyz="-0.022933 0.001906 -0.017233"/>
		</collision>
		<inertial>
			<mass value="0.005332"/>
			<inertia ixx="0.000015" ixy="0.000001" ixz="-0.000004" iyy="0.000017" iyz="0.000004" izz="0.000006"/>
			<origin rpy="0 0 0" xyz="-0.029511 0.002253 -0.023968"/>
		</inertial>
	</link>
	<joint name="r_wrist_x" type="revolute">
		<parent link="right_wrist:11"/>
		<child link="right_palm:11"/>
		<origin rpy="-0.433097 0 0" xyz="0.023416 0.006052 -0.070337"/>
		<axis xyz="1 0 0"/>
		<limit effort="30" lower="0" upper="0.872665" velocity="1"/>
	</joint>
	<link name="right_tcp">
		<origin rpy="0 0 0" xyz="0 0 0"/>
	</link>
	<joint name="r_grasp_point" type="fixed">
		<parent link="right_palm:11"/>
		<child link="right_tcp"/>
		<!-- Relative to wrist x joint
		Coppeliasim aligns z with rotation axis, URDF has it at x, so frame is rotated 90 degrees around y.
		Coppeliasim x y z in parent frame (relative to wrist_x) is actually z y -x here -->
		<origin rpy="0 1.7802358370342162 1.57079633" xyz="-2.4266e-02 4.4322e-02 -3.4591e-02"/>
	</joint>
</robot>
